### PR TITLE
Write all field keys in lower case

### PR DESCRIPTION
**Pull request by koppor**
Write all field keys in lower case. No more camel casing of field names. E.g., `title` is written instead of `Title`, `howpublished` instead of `HowPublished`, and `doi` instead of `DOI`. The configuration option `Use camel case for field names (e.g., "HowPublished" instead of "howpublished")` is gone.

Refs #116

### CODE REVIEW

File: src/main/java/net/sf/jabref/bibtex/CamelCaser.java [hunk 1]
```diff
@@ -1,101 +0,0 @@
-package net.sf.jabref.bibtex;
-
-import java.util.HashMap;
-import java.util.Map;
-
-class CamelCaser {
```
Comment (lenhard) [inline src/main/java/net/sf/jabref/bibtex/CamelCaser.java]: That class might have been useful as a future formatter, but nevermind. We can rewrite it if needed.

File: src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java [hunk 2]
```diff
@@ -53,24 +51,28 @@ public void testSerialization() throws IOException {
...
-                "  Author                   = {Foo Bar}," + Globals.NEWLINE +
-                "  Journal                  = {International Journal of Something}," + Globals.NEWLINE +
-                "  Note                     = {some note}," + Globals.NEWLINE +
-                "  Number                   = {1}" + Globals.NEWLINE +
+                "  author                   = {Foo Bar}," + Globals.NEWLINE +
+                "  journal                  = {International Journal of Something}," + Globals.NEWLINE +
+                "  note                     = {some note}," + Globals.NEWLINE +
+                "  number                   = {1}" + Globals.NEWLINE +
                 "}";
+        // @formatter:on
```
Comment (lenhard) [inline src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java]: Please do not simply rewrite the tests this way! I used this test to ensure that a rewrite without modification does not change the serialization. Your change eliminates this information! Please only change the test that modify an entry (which results in a reformatting on write)

File: src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java [hunk 3]
```diff
@@ -95,12 +97,14 @@ public void roundTripTest() throws IOException {
...
```
Comment (lenhard) [inline src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java]: See above.

File: src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java [hunk 4]
```diff
@@ -151,24 +157,28 @@ public void roundTripWithModification() throws IOException {
...
         assertEquals(expected, actual);
     }
 
     @Test
     public void roundTripWithCamelCasing() throws IOException {
+        // @formatter:off
```
Comment (lenhard) [inline src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java]: If camel casing is turned off, you should propably also rename this method.

File: src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java [hunk 5]
```diff
@@ -190,24 +200,28 @@ public void roundTripWithCamelCasing() throws IOException {
...
-                "  Author                   = {Foo Bar}," + Globals.NEWLINE +
-                "  Journal                  = {International Journal of Something}," + Globals.NEWLINE +
-                "  Note                     = {some note}," + Globals.NEWLINE +
-                "  Number                   = {1}" + Globals.NEWLINE +
+                "  author                   = {Foo Bar}," + Globals.NEWLINE +
+                "  journal                  = {International Journal of Something}," + Globals.NEWLINE +
+                "  note                     = {some note}," + Globals.NEWLINE +
+                "  number                   = {1}" + Globals.NEWLINE +
                 "}\n\n";
+        // @formatter:on
```
Comment (lenhard) [inline src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java]: See above. Please do not modify casing here!

File: src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java [hunk 6]
```diff
@@ -233,12 +247,14 @@ public void roundTripWithAppendedNewlines() throws IOException {
...
-                "  Number                   = {1}" + Globals.NEWLINE +
+                "  author                   = {Foo Bar}," + Globals.NEWLINE +
+                "  journal                  = {International Journal of Something}," + Globals.NEWLINE +
+                "  note                     = {some note}," + Globals.NEWLINE +
+                "  number                   = {1}" + Globals.NEWLINE +
                 "}";
+        // @formatter:on
 
         String result = testSingleWrite(bibtexEntry);
         result = testSingleWrite(result);
```
Comment (lenhard) [inline src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java]: See above. please do not modify casing here!

File: src/main/java/net/sf/jabref/bibtex/CamelCaser.java (hunk 1 above)
Comment (koppor) [inline src/main/java/net/sf/jabref/bibtex/CamelCaser.java]: I also found the class a good example for good code. However, we decided to remove unused code. :neckbeard:

File: src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java (hunk 2 above)
Comment (koppor) [inline src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java]: The current PR changes the serialization. All fields will be written lower case. So I have to change `roundTripTest`, don't I?

File: src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java (hunk 4 above)
Comment (koppor) [inline src/test/java/net/sf/jabref/bibtex/BibEntryWriterTest.java]: Done. `roundTripWithCamelCasingInTheOriginalEntryAndResultInLowerCase`

**Comment by koppor** [comment]
Maybe, the UI also has to display lower case values or should it use the (currently removed) CamelCase functionality? I don't like the current `EntryUtil.capitalizeFirst` solution (found in MainTableColumn.java).

**Comment by lenhard** [comment]
In my opinion, it should use lower case, since we are standardizing on that. I'd say you are free to implement what you find more beautiful :)

**Comment by simonharrer** [comment]
:+1: on lower case.

**Pull request merged**